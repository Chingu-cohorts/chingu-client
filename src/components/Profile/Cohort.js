import React from "react";

import "./Cohort.css";

const Cohort = props => (
  <div className="cohort">
    <h4>{props.name}</h4>
    <p>
      {props.members} Members
    </p>
  </div>
);

export default Cohort;
