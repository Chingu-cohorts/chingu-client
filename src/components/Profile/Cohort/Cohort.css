.cohort {
  background: #f5f5f5;
  padding: 1rem;
  margin-bottom: 1rem;
}

.cohort h4 {
  font-size: 1.2rem;
  font-weight: 600;
}

.cohort p {
  font-size: 1rem;
}
