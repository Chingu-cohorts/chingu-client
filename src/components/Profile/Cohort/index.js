import React from "react";

import "./Cohort.css";

const Cohort = props => (
  <div className="col s12 m12 grey lighten-4 cohort">
    <h5>{props.name}</h5>
    <p className="grey-text lighten-1">
      {props.members} Members, {props.countries} Countries, {props.projects}{" "}
      Projects
    </p>
  </div>
);

export default Cohort;
