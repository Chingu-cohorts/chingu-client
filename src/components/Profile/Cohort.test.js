import React from "react";
import { shallow } from "enzyme";

import Cohort from "./Cohort";

describe("Cohort Component", () => {
  const props = {
    name: "Voyage X",
    members: 1000
  };

  it("renders without crashing", () => {
    const wrapper = shallow(<Cohort {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("has a cohort class", () => {
    const wrapper = shallow(<Cohort {...props} />);
    expect(wrapper.hasClass("cohort")).toEqual(true);
  });

  it("contains an h4 with class cohort-name", () => {
    const wrapper = shallow(<Cohort {...props} />);
    expect(wrapper.find("h4.cohort-name")).toHaveLength(1);
  });

  it("contains a p with class cohort-details", () => {
    const wrapper = shallow(<Cohort {...props} />);
    expect(wrapper.find("p.cohort-details")).toHaveLength(1);
  });
});
