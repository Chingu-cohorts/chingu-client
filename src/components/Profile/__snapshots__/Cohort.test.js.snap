// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cohort
    teams={
        Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ]
    }
    title="Voyage X"
    users={
        Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="cohort-name"
>
          Voyage X
</h4>,
        <p
          className="cohort-details"
>
          2
           Members, 
          2
           Teams
</p>,
      ],
      "className": "cohort",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Voyage X",
          "className": "cohort-name",
        },
        "ref": null,
        "rendered": "Voyage X",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            2,
            " Members, ",
            2,
            " Teams",
          ],
          "className": "cohort-details",
        },
        "ref": null,
        "rendered": Array [
          2,
          " Members, ",
          2,
          " Teams",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="cohort-name"
>
            Voyage X
</h4>,
          <p
            className="cohort-details"
>
            2
             Members, 
            2
             Teams
</p>,
        ],
        "className": "cohort",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Voyage X",
            "className": "cohort-name",
          },
          "ref": null,
          "rendered": "Voyage X",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              2,
              " Members, ",
              2,
              " Teams",
            ],
            "className": "cohort-details",
          },
          "ref": null,
          "rendered": Array [
            2,
            " Members, ",
            2,
            " Teams",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
